kyo-core/shared/src/main/scala/kyo/typeMap.scala

```diff
@@ -12,22 +12,13 @@ object TypeMap:
             throw new RuntimeException(s"fatal: kyo.TypeMap of contents [${self.show}] missing value of type: [${t.showTpe}].")
 
         def get[B >: A](using t: Tag[B]): B =
-            val b = self.getOrElse(t.erased, null)
-            if !isNull(b) then b.asInstanceOf[B]
-            else
-                var sub: B = null.asInstanceOf[B]
-                val it     = self.iterator
-                try // iterator should never throw given type constraint on B
-                    while isNull(sub) do
-                        val (tag, item) = it.next()
-                        if tag <:< t then
-                            sub = item.asInstanceOf[B]
-                    end while
-                catch
-                    case _: NoSuchElementException => fatal
-                end try
-                sub
-            end if
+            val it = self.iterator
```

Comment:
would it be possible to keep the exact lookup first? The linear search on every `get` seems expensive.